scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,172 @@
+            bytecodeOffsetField.setAccessible(true);
+            FLAG_RESOLVED_FIELD.setAccessible(true);
+            int FLAG_RESOLVED = (Integer) FLAG_RESOLVED_FIELD.get(null);
+
+            short flags = (Short) flagsField.get(startLabel);
+            flags |= FLAG_RESOLVED;
+            flagsField.set(startLabel, flags);
+            bytecodeOffsetField.set(startLabel, scopeStart);
+
+            flags = (Short) flagsField.get(endLabel);
+            flags |= FLAG_RESOLVED;
+            flagsField.set(endLabel, flags);
+            bytecodeOffsetField.set(endLabel, scopeStart + scopeLength);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        this.start = new Label[] {startLabel};
+        this.end = new Label[] {endLabel};
+        this.index = new int[] {index};
+        this.variableName = null;
     }
 
+    /**
+     * Returns the bytecode offset to the start of the first scope/lifetime.
+     *
+     * @return the bytecode offset to the start of the first scope/lifetime
+     */
+    public int getScopeStart() {
+        try {
+            return start[0].getOffset();
+        } catch (IllegalStateException e) {
+            System.err.println("Labels not resolved: " + Arrays.toString(start));
+            return -1;
+        }
+    }
 
+    // This is used only in IndexFileWriter.
     /**
-     * Returns whether this {@link LocalLocation} equals <code>o</code>; a
-     * slightly faster variant of {@link #equals(Object)} for when the argument
-     * is statically known to be another nonnull {@link LocalLocation}.
+     * Returns the length of all the scopes/lifetimes (in bytes).
+     *
+     * @return the length of all the scopes/lifetimes (in bytes)
      */
-    public boolean equals(LocalLocation l) {
-        return index == l.index && scopeStart == l.scopeStart
-                && scopeLength == l.scopeLength &&
-                (varName==null || varName.equals(l.varName)) &&
-                varIndex==l.varIndex;
+    public int getScopeLength() {
+        try {
+            return end[end.length - 1].getOffset() - getScopeStart();
+        } catch (IllegalStateException e) {
+            System.err.println("Labels not resolved: " + Arrays.toString(end));
+            return -1;
+        }
     }
 
     /**
-     * This {@link LocalLocation} equals <code>o</code> if and only if
-     * <code>o</code> is another nonnull {@link LocalLocation} and
-     * <code>this</code> and <code>o</code> have equal {@link #index},
-     * {@link #scopeStart}, and {@link #scopeLength}.
+     * Returns the local variable index.
+     * @return the local variable index
      */
+    public int getVarIndex() {
+        return index[0];
+    }
+
     @Override
     public boolean equals(Object o) {
-        return o instanceof LocalLocation
-                && equals((LocalLocation) o);
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
This is a bit non-idiomatic.  The more typical code would be
```
if (! o instanceof LocalLocation) {
  return false;
}
```

Why does this do an exact test instead of the usual subtyping test done by `instanceof`?

I have the same question at some other occurrences of this pattern.

Reply:
I changed this equals to be similar to others in the Annotation Tools.  Where are the other occurrences you are referring to?

Reply:
I don't recall; search for `o.getClass()`